react/ShelfContent.js

```diff
@@ -1,172 +1,166 @@
 
 import shelf from './shelf.css'
 
+const SLIDER_WIDTH_ONE_ELEMENT = 320
+const SLIDER_WIDTH_TWO_ELEMENTS = 500
+const SLIDER_WIDTH_THREE_ELEMENTS = 750
+const SLIDER_WIDTH_FIVE_ELEMENTS = 1290
+const SLIDER_WIDTH_FOUR_ELEMENTS = 1000
 const DEFAULT_SHELF_ITEM_WIDTH = 260
-const DOTS_LARGE_VIEWPORT = true
-const SLIDES_TO_SCROLL_LARGE_VIEWPORT = 1
-
-const BREAKPOINT_MOBILE_VIEWPORT = 600
-const SLIDER_CENTER_MOBILE_MODE = true
-const ARROWS_MOBILE_VIEWPORT = false
-const DOTS_MOBILE_VIEWPORT = false
-const DEFAULT_ITEMS_MOBILE = 1
-const DEFAULT_ITEMS_DESKTOP = 3
-const SLIDES_TO_SCROLL_MOBILE_VIEWPORT = 1
-const SLIDES_TO_SHOW_MOBILE_VIEWPORT = 1
-
-const VARIABLE_WIDTH_MOBILE_MODE = true
-
-const BREAKPOINT_EXTRA_SMALL_MOBILE_VIEWPORT = 350
-const DOTS_EXTRA_SMALL_MOBILE_VIEWPORT = true
-const SLIDER_CENTER_MODE_EXTRA_SMALL_MOBILE = true
-
 const ITEMS_TO_FULL_WIDTH = 5
 
 /**
  * ShelfContent Component. Executes the interaction with react-slick
  * and render the properly content of the Shelf depending of edit mode state.
  */
 class ShelfContent extends Component {
-  getSliderSettings = itemsPerPage => {
-    const { arrows } = this.props
-    return {
-      slidesToShow: itemsPerPage,
-      slidesToScroll: SLIDES_TO_SCROLL_LARGE_VIEWPORT,
-      dots: DOTS_LARGE_VIEWPORT,
-      arrows,
-      responsive: [
-        {
-          breakpoint: BREAKPOINT_MOBILE_VIEWPORT,
-          settings: {
-            slidesToShow: SLIDES_TO_SHOW_MOBILE_VIEWPORT,
-            slidesToScroll: SLIDES_TO_SCROLL_MOBILE_VIEWPORT,
-            arrows: ARROWS_MOBILE_VIEWPORT,
-            dots: DOTS_MOBILE_VIEWPORT,
-            centerMode: SLIDER_CENTER_MOBILE_MODE,
-            variableWidth: VARIABLE_WIDTH_MOBILE_MODE,
-          },
-        },
-        {
-          breakpoint: BREAKPOINT_EXTRA_SMALL_MOBILE_VIEWPORT,
-          settings: {
-            slidesToShow: SLIDES_TO_SHOW_MOBILE_VIEWPORT,
-            slidesToScroll: SLIDES_TO_SCROLL_MOBILE_VIEWPORT,
-            arrows: ARROWS_MOBILE_VIEWPORT,
-            dots: DOTS_EXTRA_SMALL_MOBILE_VIEWPORT,
-            centerMode: SLIDER_CENTER_MODE_EXTRA_SMALL_MOBILE,
-            variableWidth: VARIABLE_WIDTH_MOBILE_MODE,
-          },
-        },
-      ],
+  constructor(props) {
+    super(props)
+    this.perPage = {
+      [SLIDER_WIDTH_FIVE_ELEMENTS]: 5,
+      [SLIDER_WIDTH_FOUR_ELEMENTS]: 4,
+      [SLIDER_WIDTH_THREE_ELEMENTS]: 3,
+      [SLIDER_WIDTH_TWO_ELEMENTS]: 2,
+      [SLIDER_WIDTH_ONE_ELEMENT]: 1,
+    }
+    this.calcItemsPerPage()
+    this.state = {
+      currentSlide: 0,
+      firstRender: true,
     }
   }
 
-  getClassByItemsPerPage(itemsPerPage) {
-    switch (itemsPerPage) {
-      case 5:
-        return 'w-20'
-      case 4:
-        return 'w-25'
-      case 3:
-        return 'w-third'
+  calcItemsPerPage = () => {
+    const { itemsPerPage } = this.props
+    for (var key in this.perPage) {
+      this.perPage[key] > itemsPerPage && delete this.perPage[key]
```

Comment:
I think would be easier do understand if you make aa `if` statement instead of making a single expression.